common/views/components/IIIFPresentationPreview/IIIFPresentationPreview.js

```diff
@@ -1,199 +1,156 @@
 // @flow
+import fetch from 'isomorphic-unfetch';
 import { iiifImageTemplate } from '@weco/common/utils/convert-image-uri';
-import { Fragment } from 'react';
-import styled from 'styled-components';
-import { classNames } from '../../../utils/classnames';
-import Button from '@weco/common/views/components/Buttons/Button/Button';
+import { Fragment, useEffect, useState } from 'react';
+import { type iiifPresentationLocation } from '@weco/common/utils/works';
 
-const BookContainer = styled.div`
-  display: inline-block;
-  position: relative;
-  z-index: 1;
-  margin-bottom: 20px;
-  word-spacing: 0;
+// Ideal preview thumbnails order: Title page, Front Cover, first page of Table of Contents, 2 random.
+// If we don't have any of the sructured pages, we fill with random ones, so there are always 5 images if possible.
+function randomImages(iiifManifest = null, structuredImages = [], n = 1) {
```

Comment:
When is `iiifManifest` not sent?
Should this be dealt with outside the function?

Reply:
Hmm, but confused. I'd already removed this: https://github.com/wellcometrust/wellcomecollection.org/pull/4208/commits/40f825be4974dea0646eb8dc90ba301ed4457911#diff-870b0e72f1025374211ac9ef40bebbd7L107